tensorboard/data/server/run.rs

```diff
@@ -525,4 +547,76 @@ mod test {
 
         Ok(())
     }
+
+    #[test]
+    fn test_plugin_sampling_hint() -> Result<(), Box<dyn std::error::Error>> {
+        use std::str::FromStr;
+
+        let logdir = tempfile::tempdir()?;
+        let f1_name = logdir.path().join("tfevents.123");
+        let mut f1 = BufWriter::new(File::create(&f1_name)?);
+
+        // Write some data points.
+        let run = Run("train".to_string());
+        let tag = Tag("accuracy".to_string());
+        f1.write_scalar(&tag, Step(0), WallTime::new(1235.0).unwrap(), 0.25)?;
+        f1.write_scalar(&tag, Step(1), WallTime::new(1236.0).unwrap(), 0.50)?;
+        f1.write_scalar(&tag, Step(2), WallTime::new(1237.0).unwrap(), 0.75)?;
+        f1.write_scalar(&tag, Step(3), WallTime::new(1238.0).unwrap(), 1.00)?;
+        // Write some blobs.
+        f1.write_graph(
+            Step(0),
+            WallTime::new(1235.0).unwrap(),
+            Bytes::from_static(b"<sample model graph>"),
+        )?;
+        f1.write_graph(
+            Step(1),
+            WallTime::new(1236.0).unwrap(),
+            Bytes::from_static(b"<sample model graph>"),
+        )?;
+        f1.write_graph(
+            Step(2),
+            WallTime::new(1237.0).unwrap(),
+            Bytes::from_static(b"<sample model graph>"),
+        )?;
+        // flush, so that the data's there when we read it
+        f1.into_inner()?.sync_all()?;
+
+        let plugin_sampling_hint = PluginSamplingHint::from_str("scalars=2").unwrap();
+
+        let mut loader = RunLoader::new(run.clone(), Arc::new(Some(plugin_sampling_hint)));
+        let logdir = DiskLogdir::new(logdir.path().to_path_buf());
+        let commit = Commit::new();
+        commit
+            .runs
+            .write()
+            .expect("write-locking runs map")
+            .insert(run.clone(), Default::default());
+        loader.reload(
+            &logdir,
+            vec![EventFileBuf(f1_name)],
+            &commit.runs.read().unwrap()[&run],
+        );
+
+        let runs = commit.runs.read().expect("read-locking runs map");
+        let run_data: &commit::RunData = &*runs
+            .get(&run)
+            .expect("looking up data for run")
+            .read()
+            .expect("read-locking run data map");
+
+        let scalar_ts = run_data.scalars.get(&tag).unwrap();
+        let scalar_ts_values = scalar_ts.valid_values().collect::<Vec<_>>();
```

Comment:
Since you only care about the count, you can use `.count()` instead of
`.collect()`.